src/ir_def/construct.ml

```diff
@@ -104,6 +104,7 @@ let primE prim es =
     | SerializePrim _ -> T.blob
     | SystemCyclesAvailablePrim
     | SystemCyclesAcceptPrim -> T.nat
+    | DeserializePrim ts -> T.Tup ts
```

Comment:
```suggestion
    | DeserializePrim ts -> T.seq ts
```

Reply:
T.seq avoids creating the one-tuple.

Reply:
I think that this was the bug you asked about on slack.

Reply:
This overcomes the type mismatch in the IR, yes.

However, now I get another error, since morally, we *are* encoding a 1-tuple when we invoke `to_candid "Foo"`.

Namely, `IDL error: unexpected IDL type when parsing Text`, which I suspect could be because the candid for `Text` and `(Text,)` are not the same?

Do I need to also special-case the `to_candid` desugaring to avoid 1-tuples there?

I had a version like that when I was experimenting, but it seemed too complex to case analyze for that.  Now I wonder if that's what you had imagined?

Reply:
Or maybe the other suggestion compensates:
https://github.com/dfinity/motoko/pull/3155/files#r860161312

Let me try.

Reply:
Nevermind.  I tried some other variations of the test program.

The deserialization issue that I see now is not specific to 1-tuples.  It also happens for the `(Nat, Nat, Nat)` case:

```
+IDL error: too few arguments NNN
+Error: failed to run main module `_out/idl.wasm`
+
+Caused by:
+    0: failed to invoke command default
+    1: wasm trap: unreachable
+       wasm backtrace:
+         0: @deserialize<NNN>
+         1: deserNats
+         2: init
+         3: _start
```

Will push my code now.